File: my-submissions/0056-merge-intervals/solution.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end()); //sort as per the start time intervals
        int k = 0;  //index for merged intervals
        for(int i = 1;i<intervals.size();i++){
            if(intervals[k][1] >= intervals[i][0]){
                intervals[k][1] = max(intervals[k][1],intervals[i] [1]); //store that element end intervals which is greater on that merging elements
            }
            else
            {
                k++; //move to next index
                intervals[k] = intervals[i]; // shifting that k index to that intervals in which the +1 interval may be merge
            }
        }
            intervals.resize(k+1);
            return intervals;
    }
};

File: my-submissions/0056-merge-intervals/solution.java
class Solution {
    public int[][] merge(int[][] intervals) {
        int n = intervals.length;

        if(n <= 1) return intervals;

        Arrays.sort(intervals, (a,b) -> Integer.compare(a[0],b[0]));

        List<List<Integer>> li = new ArrayList<>();
        li.add(Arrays.asList(intervals[0][0],intervals[0][1]));

        int i = 1;
        while(i<n){
            int start = intervals[i][0];
            int end = intervals[i][1];

            if(li.get(li.size() - 1).get(1) >= start ){
                List<Integer> row = li.get(li.size() - 1);
                row.set(1,Math.max(end,li.get(li.size() - 1).get(1)));
            }
            else{
                li.add(Arrays.asList(intervals[i][0],intervals[i][1]));
            }
            i++;
        }
        int[][] ans = new int[li.size()][2];
        for(i = 0;i<li.size();i++){
            ans[i][0] = li.get(i).get(0);
            ans[i][1] = li.get(i).get(1);
        }
        return ans;
    }
}
